sklearn/preprocessing/tests/test_data.py

```diff
@@ -411,6 +411,15 @@ def test_scale_sparse_with_mean_raise_exception():
     assert_raises(ValueError, scaler.inverse_transform, X_transformed_csr)
 
 
+def test_scale_input_finiteness_validation():
+    """Check if non finite inputs raise ValueError"""
+    X = [np.nan, 5, 6, 7, 8]
+    assert_raises(ValueError, scale, X)
+
+    X = [np.inf, 5, 6, 7, 8]
+    assert_raises(ValueError, scale, X)
+
+
```

Comment:
Can you please use `assert_raises_regex` to check for the presence of some relevant key word or phrase in the exception message?